lib/mixpanel-node.js

```diff
@@ -148,16 +148,24 @@ var create_client = function(token, config) {
             }
             try {
                 body = JSON.parse(body);
-                if (body.status === 0) {
-                    error = new Error('Mixpanel REST API reported an error: ' + body.error);
-                    console.error(error);
-                    return callback(error);
-                }
             } catch (e) {
-                error = new Error('Mixpanel REST API returned invalid JSON response');
+                error = new Error('Mixpanel REST API returned an invalid JSON response');
+                console.error(error);
+                return callback(error);
+            }
+
+            if (!body || typeof body.status !== 'number') {
```

Comment:
What if the body status is 2? what do you return here? `if (body && body.status === 1)` will most of the time be true, so it would not call the else, if you don't have to make 2 iff-s it's easy
```
if (body) {
  if (body.status === 1) {
    return callback
  }
  if (body.status === 0 {
    return callback(error)
  }
}
return callback(invalid error)
```